autotest/gdrivers/fits.py

```diff
@@ -82,3 +81,34 @@ def fits_metadata():
     ds2 = None
 
     assert md['TEST2'] == 'test_value2'
+
+def test_fits_nodata():
+    driver = gdal.GetDriverByName('FITS')
+
+    ds = gdal.Open('../gcore/data/nodata_byte.tif')
+    ds2 = driver.CreateCopy('tmp/nodata_byte.fits', ds)
+    ds2 = None
+    gdal.Unlink('tmp/nodata_byte.fits.aux.xml')
+
+    ds2 = gdal.Open('tmp/nodata_byte.fits')
```

Comment:
the file should be Unlink() after use

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/8e48b80f479616a461e1cbe590cdd4e32baed012